js/ui/windowManager.js

```diff
@@ -717,6 +784,126 @@ const WindowManager = new Lang.Class({
+    },
+    
+    _updateReadyRotateAnimationsWith: function (window) {
+        /* A new window was added. Get its pid and look for any
+         * unsatisfied entries in _pendingRotateAnimations */
+        const pid = window ? window.get_meta_window().get_pid() : null;
+        const lastPendingRotateAnimationsLength = this._pendingRotateAnimations.length;
+        this._pendingRotateAnimations = this._pendingRotateAnimations.filter(Lang.bind(this, function(animationSpec) {
+            let unsatisfiedPids = 0;
+            Object.keys(animationSpec).forEach(function(key) {
+                if (animationSpec[key].window == null) {
+                    if (animationSpec[key].pid == pid) {
+                        animationSpec[key].window = window;
+                    } else {
+                        unsatisfiedPids++;
+                    }
+                }
+            });
+            
+            if (unsatisfiedPids == 0) {
+                /* Delaying the animation like this is not ideal, but it is necessary
+                 * to allow some time for the window to paint itself for the first time.
+                 *
+                 * Mutter doesn't seem to have any kind of notification for when we first
+                 * receive a damage event on the window - that will be the time when the bound
+                 * texture will be defined. Until that point, the contents are just empty and
+                 * so we can't just rely on map events to get to where we want. */
+                animationSpec.dst.window.rotation_angle_y = -180;
+                animationSpec.dst.window.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+                animationSpec.src.window.pivot_point = new Clutter.Point({ x: 0.5, y: 0.5 });
+                animationSpec.src.window.add_effect_with_name('enable-culling',
+                                                              new EnableBackfaceCullingEffect());
+                animationSpec.dst.window.add_effect_with_name('enable-culling',
+                                                              new EnableBackfaceCullingEffect());
+                animationSpec.dst.window['opacity'] = 0;
+                let dst_geometry = animationSpec.src.rect;
+                animationSpec.dst.window.get_meta_window().move_resize_frame(false,
+                                                                             dst_geometry.x,
+                                                                             dst_geometry.y,
+                                                                             dst_geometry.width,
+                                                                             dst_geometry.height);
+
+                this._rotateInActors.push(animationSpec.dst.window);
+                this._rotateOutActors.push(animationSpec.src.window);
+
+                let waitingRotateTimeout = GLib.timeout_add_seconds(GLib.PRIORITY_DEFAULT, 1, Lang.bind(this, function() {
```

Comment:
I've noticed that builder seems to take quite some time before it first draws the window and one second seems to be the amount of time I need to wait before it is usually ready on my machine. On other machines it could be slower.

I'm not really sure about the best way of dealing with this. Digging around mutter, it seems like there doesn't seem to be any tracking of when a window was first damaged (which is technically when the pixmap contents are first defined).